cmd/tailscale/cli/file.go

```diff
@@ -387,54 +389,45 @@ func receiveFile(ctx context.Context, wf apitype.WaitingFile, dir string) (targe
 	if err != nil {
 		return "", 0, fmt.Errorf("opening inbox file %q: %w", wf.Name, err)
 	}
+	defer rc.Close()
 	f, err := openFileOrSubstitute(dir, wf.Name, getArgs.conflict)
 	if err != nil {
 		return "", 0, err
 	}
 	_, err = io.Copy(f, rc)
-	rc.Close()
 	if err != nil {
+		f.Close()
 		return "", 0, fmt.Errorf("failed to write %v: %v", f.Name(), err)
 	}
 	return f.Name(), size, f.Close()
 }
 
-func runFileGet(ctx context.Context, args []string) error {
-	if len(args) != 1 {
-		return errors.New("usage: file get <target-directory>")
-	}
-	log.SetFlags(0)
-
-	dir := args[0]
-	if dir == "/dev/null" {
-		return wipeInbox(ctx)
-	}
-
-	if fi, err := os.Stat(dir); err != nil || !fi.IsDir() {
-		return fmt.Errorf("%q is not a directory", dir)
-	}
-
+func runFileGetOneBatch(ctx context.Context, dir string) []error {
 	var wfs []apitype.WaitingFile
 	var err error
-	for {
+	var errs []error
+	for len(errs) < 2 {
```

Comment:
why 2? add a little comment saying what's special about this number?

Reply:
converted to 0, which I think is correct having read the code i was calling.

i had thought `GetWaitingFile` might return an error if there were no files waiting. (1 error), and I wanted the loop to go around at least once through `WaitForFiles` and a `GetWaitingFile` re-attempt if there were initially no files waiting.  looking at the code, that's not how `GetWaitingFile` works.